app/routes/profile.tsx

```diff
@@ -0,0 +1,92 @@
+import { json, redirect, Session } from "@remix-run/node";
+import type { LoaderFunction } from "@remix-run/node";
+import type { FC } from "react";
+import { commitSession, destroySession, getSession } from "~/session.server";
+import { useLoaderData } from "@remix-run/react";
+import RootLayout from "~/components/RootLayout";
+import { auth } from "~/firebase.server";
+
+/**
+ * Returns the user that's currently logged in, or null if there is none.
+ *
+ * @see https://firebase.google.com/docs/auth/admin/verify-id-tokens#web
+ */
+export async function getCurrentUser(session: Session) {
+  const idToken = session.get("idToken");
+
+  const error = session.get("error");
+  if (error) {
+    throw new Error(error);
+  }
+
+  if (typeof idToken !== "string") {
+    return null;
+  }
+
+  try {
+    const decodedClaims = await auth.verifyIdToken(idToken);
+    return decodedClaims;
+  } catch (err) {
+    // The session isn't valid anymore so return undefined
+    return null;
+  }
+}
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("Cookie"));
+  const currentUser = await getCurrentUser(session);
+
+  if (!currentUser) {
+    // Redirect to the login page if needed
+    session.unset("idToken");
+    return redirect("/login", {
+      headers: {
+        "Set-Cookie": await destroySession(session),
+      },
+    });
+  }
+
+  const data: User = {
+    displayName: currentUser.name,
+    avatar: currentUser.picture,
+    email: currentUser.email,
+  };
+
+  return json(data, {
+    headers: { "Set-Cookie": await commitSession(session) },
+  });
+};
+
+type User = {
+  displayName: string;
+  avatar?: string;
+  email?: string;
+};
+
+const Profile: FC = () => {
```

Comment:
Here's the profile page: all it does is display the user's avatar and name, nothing fancy. But where does that data come from?